core/src/main/java/org/apache/calcite/rel/rules/AggregateExpandDistinctAggregatesRule.java

```diff
@@ -151,6 +167,19 @@ public void onMatch(RelOptRuleCall call) {
       return;
     }
 
+    // If only one distinct aggregate and one or more non-distinct aggregates,
+    // we can generate multi-phase aggregates
+    if (distinctCount == 1                // one distinct aggregate
+            && !hasGroupingSets           // no grouping sets
+            && !hasUnsupportedAggFunc     // sum/min/max/count in non-distinct aggregate
+            && nonDistinctCount > 0       // one or more non-distinct aggregates
+            && aggregate.getGroupSet().size() > 0) { // at least one group-by key present
```

Comment:
Sorry, I did not understand that. Could you please explain with the example above assuming you mean a SQL CAST?
{source}
select count(*), count(distinct deptno) from emp
{source}
would transform to  
{source}
select sum(cnt), count(deptno)
from (select count(*) as cnt, deptno 
         from emp
         group by deptno)
{source}
but will fail nullability checks. 
What should the CAST form look like?